### ISSUE TITLE
What is the use of Hessian for custom objective?

### ISSUE THREAD

**Issue by liujxing** [Microsoft/LightGBM#1463]
I read the paper of LightGBM and it seems the splitting is chosen by the criterion of variance gain calculated using gradient, so where does the Hessian defined in custom objective come into play here?

**Comment by guolinke** [comment]
gradient is the first order optimization. 
gradient with hessian is the second order optimization, like the newton optimization.
Refer to https://xgboost.readthedocs.io/en/latest/model.html

**Comment by guolinke** [comment]
just replace the count with hessian, and the variance gain becomes to the hessian optimization.

**Comment by liujxing** [comment]
Thanks.
